src/correction/cellbender_remove_background/helper.py

```diff
@@ -0,0 +1,137 @@
+import tables
+import numpy as np
+import scipy.sparse as sp
+import anndata
+from typing import Dict
+
+
+def anndata_from_h5(file: str,
+                    analyzed_barcodes_only: bool = True) -> 'anndata.AnnData':
+    """Load an output h5 file into an AnnData object for downstream work.
+
+    Args:
+        file: The h5 file
+        analyzed_barcodes_only: False to load all barcodes, so that the size of
+            the AnnData object will match the size of the input raw count matrix.
+            True to load a limited set of barcodes: only those analyzed by the
+            algorithm. This allows relevant latent variables to be loaded
+            properly into adata.obs and adata.obsm, rather than adata.uns.
+
+    Returns:
+        adata: The anndata object, populated with inferred latent variables
+            and metadata.
+
+    """
+
+    d = dict_from_h5(file)
+    X = sp.csc_matrix((d.pop('data'), d.pop('indices'), d.pop('indptr')),
+                      shape=d.pop('shape')).transpose().tocsr()
+
+    # check and see if we have barcode index annotations, and if the file is filtered
+    barcode_key = [k for k in d.keys() if (('barcode' in k) and ('ind' in k))]
+    if len(barcode_key) > 0:
+        max_barcode_ind = d[barcode_key[0]].max()
+        filtered_file = (max_barcode_ind >= X.shape[0])
+    else:
+        filtered_file = True
+
+    if analyzed_barcodes_only:
+        if filtered_file:
+            # filtered file being read, so we don't need to subset
+            print('Assuming we are loading a "filtered" file that contains only cells.')
+            pass
+        elif 'barcode_indices_for_latents' in d.keys():
+            X = X[d['barcode_indices_for_latents'], :]
+            d['barcodes'] = d['barcodes'][d['barcode_indices_for_latents']]
+        elif 'barcodes_analyzed_inds' in d.keys():
+            X = X[d['barcodes_analyzed_inds'], :]
+            d['barcodes'] = d['barcodes'][d['barcodes_analyzed_inds']]
+        else:
+            print('Warning: analyzed_barcodes_only=True, but the key '
+                  '"barcodes_analyzed_inds" or "barcode_indices_for_latents" '
+                  'is missing from the h5 file. '
+                  'Will output all barcodes, and proceed as if '
+                  'analyzed_barcodes_only=False')
+
+    # Construct the anndata object.
+    adata = anndata.AnnData(X=X,
+                            obs={'barcode': d.pop('barcodes').astype(str)},
+                            var={'gene_name': (d.pop('gene_names') if 'gene_names' in d.keys()
+                                               else d.pop('name')).astype(str)},
+                            dtype=X.dtype)
+    adata.obs.set_index('barcode', inplace=True)
+    adata.var.set_index('gene_name', inplace=True)
+
+    # For CellRanger v2 legacy format, "gene_ids" was called "genes"... rename this
+    if 'genes' in d.keys():
```

Comment:
Can this be made to (not sure if the key 'id' is needed or if it is only used to rename) and if one of 'genes' or 'id' is always present

```python
d['gene_id'] = d.pop('id', d.pop('genes', None))
```